### PR TITLE
refactor(core): Move webhook DB access to repository (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

**Review by krynble — APPROVED** [review]
I have mixed feelings about using the built-in functions (such as `findById`) but since it's not something we could remove either, I think this is fine. Thanks for addressing this.

To clarify: I think having specific functions that return specific objects is best as we can optimize and make sure the desired behaviors are implemented. The usage of `findById` and such functions is too flexible and usually would lead to poorly performing queries. Not a big deal for this entity, so this is all right, but when you have joins (such as the execution repo) I think dedicated functions are best.

**Review by krynble — CHANGES_REQUESTED** [review]
I'm having problems starting n8n with this

**Comment by krynble** [comment]
To clarify:

```
    ServiceNotFoundError: Service with "MaybeConstructable<DataSource>" identifier was not found in the 
    container. Register it before usage via explicitly calling the "Container.set" function or using the 
    "@Service()" decorator.

```

I think we're having issues starting n8n and injecting the service properly. I remember having this when implementing the Execution repo. The solution was to use `Container.get` instead of injecting the class at startup.

File: packages/cli/src/ActiveWorkflowRunner.ts [hunk 1]
```diff
@@ -87,6 +88,7 @@ export class ActiveWorkflowRunner {
 		private activeExecutions: ActiveExecutions,
 		private externalHooks: ExternalHooks,
 		private nodeTypes: NodeTypes,
+		private webhookRepository: WebhookRepository,
```
Comment (krynble) [inline packages/cli/src/ActiveWorkflowRunner.ts:91]: The dependency injection here is failing at start time

**Review by netroy — COMMENTED** [review]
I've cherry-picked the fix for this from another PR. This PR should be good for review now.

**Review by krynble — APPROVED** [review]
Thanks!

**Comment by janober** [comment]
Got released with [`n8n@1.1.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.1.0)

**Pull request merged**